shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -1221,15 +1220,22 @@ impl KafkaSinkCluster {
                             .unwrap_or(true)
                     })
                     .unwrap();
+
                 find_coordinator.host = shotover_node.address.host.clone();
                 find_coordinator.port = shotover_node.address.port;
                 find_coordinator.node_id = shotover_node.broker_id;
             }
         } else {
-            // for version > 3 we can include as many coordinators as we want.
-            // so we include all shotover nodes in the rack of the coordinator.
-            let mut shotover_coordinators: Vec<Coordinator> = vec![];
-            for coordinator in find_coordinator.coordinators.drain(..) {
+            // For version > 3 we have to replace multiple coordinators.
+            // It may be tempting to include all shotover nodes in the rack of the coordinator, assuming the client will load balance between them.
+            // However it doesnt work like that.
+            // AFAIK there can only be one coordinator per unique `FindCoordinatorResponse::key`.
+            // In fact the java driver doesnt even support multiple coordinators of different types yet:
```

Comment:
Is this something that needs a test case, todo or open issue to catch once the driver does support multiple coordinators of a different type?

Is there a win in supporting this when the java driver does or should we just fix and move on?

Reply:
I believe the code I've written should handle multiple coordinators of different types.
If the java driver starts making use of that and my implementation is broken somehow, we'll know about it because it'll be broken.